src/core/contracts/utilities.ts

```diff
@@ -77,3 +78,46 @@ export interface LogEventData {
   transactionHash: HexString;
   blockNumber: number;
 }
+
+export const subscribeToEvent = async (
+  provider: ethers.providers.Provider,
```

Comment:
dont need to pass this in as argument

Reply:
We could do a getRequireProvider, but that would require potentially adding ConfigOps. I like passing it in and requiring it at the higher level, but we do mostly follow the ConfigOpts pattern.